iothub/device/tests/Mqtt/MqttIotHubAdapterTest.cs

```diff
@@ -111,9 +111,10 @@ public void TestAuthenticationChain()
             var passwordProvider = new Mock<IAuthorizationProvider>();
             var mqttIotHubEventHandler = new Mock<IMqttIotHubEventHandler>();
             var productInfo = new ProductInfo();
+            var pnpOptions = new PnPOptions();
             var mqttTransportSetting = new MqttTransportSettings(TransportType.Mqtt_Tcp_Only) { HasWill = false };
             var channelHandlerContext = new Mock<IChannelHandlerContext>();
-            var mqttIotHubAdapter = new MqttIotHubAdapter("deviceId", string.Empty, string.Empty, passwordProvider.Object, mqttTransportSetting, null, mqttIotHubEventHandler.Object, productInfo);
+            var mqttIotHubAdapter = new MqttIotHubAdapter("deviceId", string.Empty, string.Empty, passwordProvider.Object, mqttTransportSetting, null, mqttIotHubEventHandler.Object, productInfo, pnpOptions);
```

Comment:
What e2e can we add to validate positive and negative scenarios here?

Reply:
For E2E scenarios, one way to validate right now, is to check event hub for the telemetry data. Telemetry originating from pnp devices have a system property "dt-schema". Any validation for twin structure is still not ironed out by the service team.